### PR TITLE
MINOR: Controller should log UpdateMetadata response errors

### PR DESCRIPTION
**Pull request by hachikuji**
Create a controller event for handling UpdateMetadata responses and log a message when a response contains an error.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by ijuma — APPROVED** [review]
LGTM

File: core/src/main/scala/kafka/controller/ControllerState.scala [hunk 1]
```diff
@@ -104,7 +104,12 @@ object ControllerState {
     def value = 15
   }
 
+  case object UpdateMetadataResponseReceived extends ControllerState {
```
Comment (stanislavkozlovski) [inline core/src/main/scala/kafka/controller/ControllerState.scala:107]: Note that this adds a new metric - `UpdateMetadataResponseReceivedRateAndTimeMs`

Comment (hachikuji) [reply core/src/main/scala/kafka/controller/ControllerState.scala:107]: Good point. I'll disable it for now.

**Comment by hachikuji** [comment]
retest this please

**Pull request merged**